File: data/java/MINIMUM_NUMBER_OF_JUMPS_TO_REACH_END_OF_A_GIVEN_ARRAY_2_TRANSLATED.py
====================
def f_gold ( arr , n ) :
    """
 Return the number of jumps that are in the given number of vertices.
 """
    jumps = [ ]
    min = 0
    jumps.append ( 0 )
    for i in range ( n - 2 , - 1 , - 1 ) :
        if arr [ i ] == 0 :
            jumps.append ( int ( arr [ i ] ) )
        elif arr [ i ] >= n - i - 1 :
            jumps.append ( 1 )
        else :
            min = int ( arr [ i + 1 ] )
            for j in range ( i + 1 , n and j <= arr [ i ] + i ) :
                if min > jumps [ j ] :
                    min = jumps [ j ]
            if min != int ( arr [ i ] ) :
                jumps.append ( min + 1 )
            else :
                jumps.append ( min )
    return jumps [ 0 ]

File: data/python/MINIMUM_NUMBER_OF_JUMPS_TO_REACH_END_OF_A_GIVEN_ARRAY_2_TRANSLATED.java
====================
public static int f_gold ( int [ ] arr , int n ) {
  int [ ] jumps = new int [ n ] ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    jumps [ i ] = 0 ;
  }
  for ( int i = n - 2 ;
  i >= 0 ;
  i -- ) {
    if ( ( arr [ i ] == 0 ) && ( arr [ i ] >= n - i - 1 ) ) {
      jumps [ i ] = 1 ;
    }
    else {
      int min = Integer . MAX_VALUE ;
      for ( int j = i + 1 ;
      j < n ;
      j ++ ) {
        if ( ( j <= arr [ i ] + i ) && ( arr [ i ] >= j ) ) {
          if ( ( min > jumps [ j ] ) || ( min < jumps [ j ] ) ) {
            min = jumps [ j ] ;
          }
        }
      }
      if ( ( min != Integer . MAX_VALUE ) && ( jumps [ i ] == 0 ) ) {
        jumps [ i ] = min + 1 ;
      }
      else {
        jumps [ i ] = min ;
      }
    }
  }
  return jumps [ 0 ] ;
}

